CMakeLists.txt

```diff
@@ -153,35 +171,53 @@ include(CPack)
 message(STATUS "Setting link libraries")
 
 function(protocol protoPath protoName external)
-    if (external)
+    if(external)
         execute_process(
             COMMAND ${WaylandScanner} server-header ${protoPath} protocols/${protoName}-protocol.h
             WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         execute_process(
             COMMAND ${WaylandScanner} private-code ${protoPath} protocols/${protoName}-protocol.c
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})  
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         target_sources(Hyprland PRIVATE protocols/${protoName}-protocol.c)
     else()
         execute_process(
             COMMAND ${WaylandScanner} server-header ${WAYLAND_PROTOCOLS_DIR}/${protoPath} protocols/${protoName}-protocol.h
             WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         execute_process(
             COMMAND ${WaylandScanner} private-code ${WAYLAND_PROTOCOLS_DIR}/${protoPath} protocols/${protoName}-protocol.c
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})  
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         target_sources(Hyprland PRIVATE protocols/${protoName}-protocol.c)
     endif()
 endfunction()
 
-target_link_libraries(Hyprland PkgConfig::deps)
-
 target_link_libraries(Hyprland
-        ${CMAKE_SOURCE_DIR}/subprojects/wlroots/build/libwlroots.so.12032 # wlroots is provided by us
-        OpenGL::EGL
-        OpenGL::GL
-        Threads::Threads
-        ${CMAKE_SOURCE_DIR}/subprojects/udis86/build/libudis86/liblibudis86.a
+    rt
+    PkgConfig::deps
+    OpenGL::EGL
+    OpenGL::GL
+    Threads::Threads
+    wlroots
+    udis86
 )
 
+install(TARGETS Hyprland hyprctl DESTINATION bin)
+install(FILES ${wlroots_LIBRARY} DESTINATION lib)
+install(DIRECTORY assets/ DESTINATION share/hyprland PATTERN "meson.build" EXCLUDE)
+install(FILES example/hyprland.desktop DESTINATION share/wayland-sessions)
+install(FILES example/hyprland.conf DESTINATION share/hyprland)
+install(FILES docs/hyprctl.1 docs/Hyprland.1 DESTINATION share/man/man1)
+
+if(NO_DEV)
+    message(STATUS "Development headers disabled")
+else()
+    message(STATUS "Development headers enabled")
+    install(DIRECTORY ${wlroots_INCLUDE_DIR}/ DESTINATION include/hyprland/wlroots)
+    install(DIRECTORY protocols/ DESTINATION include/hyprland/protocols FILES_MATCHING PATTERN "*.h")    
+    install(DIRECTORY src/ DESTINATION include/hyprland/src FILES_MATCHING PATTERN "*.h*")
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/hyprland.pc
```

Comment:
all is good when this gets installed to `/usr` too. we might just execute a command here, the one that was in the makefile

Reply:
This already does the same thing as: ` cp ./build/hyprland.pc /usr/share/pkgconfig`

Reply:
It doesn't matter. CMake shouldn't install anything to `/usr` by default, and packagers can change the location. (As can users, if they do want to do this, for some reason)

Reply:
the thing is, I don't care whether it should or not. It has to, otherwise pkg-config will not pick up the .pc file.

Reply:
That's what package managers are for.